src/compute/src/render/reduce.rs

```diff
@@ -428,46 +430,72 @@ where
         (oks, errs.as_collection(|_, v| v.clone()))
     }
 
-    /// Build the dataflow to compute the set of distinct keys.
-    fn build_distinct<S>(
+    fn dispatch_build_distinct<S>(
         &self,
         collection: Collection<S, (Row, Row), Diff>,
     ) -> (
-        KeyValArrangement<S, Row, ()>,
+        SpecializedArrangement<S>,
         Collection<S, DataflowError, Diff>,
     )
     where
         S: Scope<Timestamp = G::Timestamp>,
+    {
+        if self.enable_specialized_arrangements {
+            let collection = collection.map(|(k, v)| {
+                assert!(v.is_empty());
+                (k, ())
+            });
```

Comment:
This check was in place for `RowRow` previously, so we should keep it in place regardless of arrangement specialization.

Reply:
Sounds good, added a `map_in_place` check on the `RowRow` variation.

Reply:
Maybe `inspect` semantically is better? It's OK to leave it as-is, but if we'd switch to columnated regions on dataflow edges, this would need to change

Reply:
Sure thing, changed to `inspect` instead.